### PR TITLE
24442 Since Magento 2.3 the wysiwyg images are not loaded anymore fro…

### LINKED ISSUE #3135: The get.php not working with database media storage
Hi guys,

I'm having a problem with my Magento 2 installation where I'm trying to use the database for media storage. I think it stems from the following function thing in `get.php`:

```
$isAllowed = function ($resource, array $allowedResources) {
    $isResourceAllowed = false;
    foreach ($allowedResources as $allowedResource) {
        if (0 === stripos($resource, $allowedResource)) {
            $isResourceAllowed = true;
        }
    }
    return $isResourceAllowed;
};
```

The `$allowedResources` array contains values like "catalog" and "wysiwyg", which are grabbed from the `system/media_storage_configuration/allowed_resources` config node.

The problem is that the `REQUEST_URI` variable for my images might be something along the lines of `/media/catalog/product/cache/1/small_image/152x190/beff4985b56e3afdbeabfc89641a4582/w/s/ws06-purple_main.jpg`. Because `/media` is appended to the start of all my request URIs for images, the `stripos($resource, $allowedResource)` check never returns 0! 

Is this a bug on Magento side or do I just have to reconfigure my Nginx configuration? It looks like you guys didn't migrate over the following line from `get.php` in Magento 1:

```
$relativeFilename = str_replace($mediaDirectory . '/', '', $pathInfo);
```

My Nginx configuration is:

```
location /media {
    try_files $uri $uri/ /get.php$is_args$args;
}
```

### LINKED ISSUE #24442: Since Magento 2.3 the wysiwyg images are not loaded anymore from media storage database.
### Preconditions (*)

1. Magento version 2.3.x

### Steps to reproduce (*)

1. Fresh install of magento with sample data
2. Synchronize media files to datase, according the manual (https://docs.magento.com/m2/ce/user_guide/system/media-storage-database.html)
3. Add following record to core_config_data table 
    - scope: default
    - scope_id: 0 
    - path: web/seo/use_rewrites
    - value: 1
4. Remove an image from media/wysiwyg editor directory
5. Image is not loading anymore from the front-end

### Expected result (*)

1. Image loaded from media database and stored to disk on webserver in media/wysiwyg

### Actual result (*)

1. First of all a 404 error occurs because media/wysiwig is not configured as allowed_resource in Magento_Cms/etc/config.xml media_storage_configuration/allowed_resources/wysiwyg_image_folder
2. File is not loaded correctly from media storage database because it's searching for media/wysiwyg/image.jpg and is stored as wysiwyg/image.jpg

### PR DESCRIPTION
**Pull request by novakivskiy**
Fixes https://github.com/magento/magento2/issues/24442
Fixes https://github.com/magento/magento2/issues/3135

### CODE REVIEW

File: app/code/Magento/MediaStorage/App/Media.php [hunk 1]
```diff
@@ -190,6 +199,17 @@ public function launch(): ResponseInterface
         return $this->response;
     }
 
+    /**
+     * Check if type image need resize
+     *
+     * @param string $originalImageName
+     * @return bool
+     */
+    public function isNeedImageResize(string $originalImageName)
```
Comment (sidolov) [inline app/code/Magento/MediaStorage/App/Media.php:208]: Should be private

Comment (sidolov) [reply app/code/Magento/MediaStorage/App/Media.php:208]: Please, add strict return type

File: app/code/Magento/MediaStorage/App/Media.php [hunk 2]
```diff
@@ -219,7 +239,8 @@ private function setPlaceholderImage(): void
      */
     private function getOriginalImage(string $resizedImagePath): string
     {
-        return preg_replace('|^.*((?:/[^/]+){3})$|', '$1', $resizedImagePath);
+        $resizedImagePath = ltrim($resizedImagePath, 'media/');
+        return preg_replace('|^catalog\/product\/cache\/[\w\d.]+(.*)$|', '$1', $resizedImagePath);
```
Comment (sidolov) [inline app/code/Magento/MediaStorage/App/Media.php:243]: Not sure that's a good idea to hardcode path in the regexp

File: app/code/Magento/MediaStorage/Service/ImageResize.php [hunk 3]
```diff
@@ -167,6 +163,20 @@ public function resizeFromImageName(string $originalImageName)
         }
     }
 
+    /**
+     * Copy image from Db Storage, if DB usage
+     *
+     * @param string $originalImageName
+     */
+    public function copyFileFromDbStorage(string $originalImageName)
```
Comment (sidolov) [inline app/code/Magento/MediaStorage/Service/ImageResize.php:171]: Should be private. 
Please, add return type

File: pub/get.php [hunk 4]
```diff
@@ -19,7 +19,7 @@
 
 $isAllowed = function ($resource, array $allowedResources) {
     foreach ($allowedResources as $allowedResource) {
-        if (0 === stripos($resource, $allowedResource)) {
+        if (0 === stripos($resource, 'media/' . $allowedResource)) {
```
Comment (sidolov) [inline pub/get.php:22]: Implementation will be broken if the resource will be configured out of `media` folder.

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by engcom-Charlie** [comment]
Hi @novakivskiy, please sign CLA and look through failed tests and requested changes. Otherwise, we can't proceed with your PR.

**Comment by sidolov** [comment]
@novakivskiy  I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**